src/backend/distributed/commands/truncate.c

```diff
@@ -67,7 +67,7 @@ ErrorIfUnsupportedTruncateStmt(TruncateStmt *truncateStatement)
 	foreach(relationCell, relationList)
 	{
 		RangeVar *rangeVar = (RangeVar *) lfirst(relationCell);
-		Oid relationId = RangeVarGetRelid(rangeVar, NoLock, true);
+		Oid relationId = RangeVarGetRelid(rangeVar, AccessShareLock, true);
```

Comment:
We could also use `RangeVarGetRelid()`, I don't know if we want to really keep relations locked at this stage?